src/Metadata.php

```diff
@@ -81,8 +81,11 @@ public function get( $check, $object_id, $meta_key, $single ) {
 				}
 			}
 		}
-		if ( empty( $check ) && $single ) {
-			$check = array( '' ); // Ensure an empty string is returned when meta is not found.
+		if ( empty( $check ) ) {
+			$check = get_metadata_default( $this->meta_type, $object_id, $meta_key, $single ); 
+			if ( $single ) {
```

Comment:
Shouldn't it be negation?

```suggestion
			if ( ! $single ) {
```